mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/nativequery/impl/NativeQueryParser.java

```diff
@@ -57,301 +67,300 @@
+				builder.setOperation( Operation.DELETEONE );
+				builder.setCriteria( cliQueryInfo.arguments.get( 0 ) );
+				if ( cliQueryInfo.arguments.size() > 1 ) {
+					builder.setOptions( cliQueryInfo.arguments.get( 1 ) );
+				}
+				break;
+			}
+			case "deleteMany": {
+				builder.setOperation( Operation.DELETEMANY );
+				builder.setCriteria( cliQueryInfo.arguments.get( 0 ) );
+				if ( cliQueryInfo.arguments.size() > 1 ) {
+					builder.setOptions( cliQueryInfo.arguments.get( 1 ) );
+				}
+				break;
+			}
+			case "update": {
+				builder.setOperation( Operation.UPDATE );
+				builder.setCriteria( cliQueryInfo.arguments.get( 0 ) );
+				builder.setUpdateOrInsert( cliQueryInfo.arguments.get( 1 ) );
+				if ( cliQueryInfo.arguments.size() > 2 ) {
+					builder.setOptions( cliQueryInfo.arguments.get( 2 ) );
+				}
+				break;
+			}
+			case "updateOne": {
+				builder.setOperation( Operation.UPDATEONE );
+				builder.setCriteria( cliQueryInfo.arguments.get( 0 ) );
+				builder.setUpdateOrInsert( cliQueryInfo.arguments.get( 1 ) );
+				if ( cliQueryInfo.arguments.size() > 2 ) {
+					builder.setOptions( cliQueryInfo.arguments.get( 2 ) );
+				}
+				break;
+			}
+			case "updateMany": {
+				builder.setOperation( Operation.UPDATEMANY );
+				builder.setCriteria( cliQueryInfo.arguments.get( 0 ) );
+				builder.setUpdateOrInsert( cliQueryInfo.arguments.get( 1 ) );
+				if ( cliQueryInfo.arguments.size() > 2 ) {
+					builder.setOptions( cliQueryInfo.arguments.get( 2 ) );
+				}
+				break;
+			}
+			case "replaceOne": {
+				builder.setOperation( Operation.REPLACEONE );
+				builder.setCriteria( cliQueryInfo.arguments.get( 0 ) );
+				builder.setUpdateOrInsert( cliQueryInfo.arguments.get( 1 ) );
+				if ( cliQueryInfo.arguments.size() > 2 ) {
+					builder.setOptions( cliQueryInfo.arguments.get( 2 ) );
+				}
+				break;
+			}
+			case "aggregate": {
+				builder.setOperation( Operation.AGGREGATE_PIPELINE );
+				Document array = Document.parse( "{ 'dummy':" + cliQueryInfo.arguments.get( 0 ) + "}" );
+				ArrayList keys = (ArrayList) array.get( "dummy" );
+				keys.forEach( key -> {
+					builder.getPipeline().add( (Document) key );
+				} );
+				break;
+			}
+			case "count": {
+				builder.setOperation( Operation.COUNT );
+				if ( cliQueryInfo.arguments.size() > 0 ) {
+					builder.setCriteria( cliQueryInfo.arguments.get( 0 ) );
+				}
+				break;
+			}
+			case "distinct": {
+				builder.setOperation( Operation.DISTINCT );
+				builder.setDistinctFieldName( JSON.parse( cliQueryInfo.arguments.get( 0 ) ).toString() );
+				if ( cliQueryInfo.arguments.size() > 1 ) {
+					builder.setCriteria( cliQueryInfo.arguments.get( 1 ) );
+				}
+				if ( cliQueryInfo.arguments.size() > 2 ) {
+					builder.setOptions( cliQueryInfo.arguments.get( 2 ) );
+				}
+				break;
+			}
+			case "mapReduce": {
+				builder.setOperation( Operation.MAP_REDUCE );
+				builder.setMapFunction( JSON.parse( cliQueryInfo.arguments.get( 0 ) ).toString() );
```

Comment:
Is there a reason why you use `JSON.parse()` for some of the options? I would have thought it would be either something to do for everything or something to avoid.

I'm also a bit curious on what sort of error we have when we have an invalid JSON here.